lib/id3c/cli/command/redcap_sync.py

```diff
@@ -0,0 +1,327 @@
+    - `warehouse.encounter`
+    - `warehouse.encounter_location`
+    - `warehouse.individual`
+    - `warehouse.location`
+    - `warehouse.sample`
+    - `warehouse.presence_absence`
+    """
+    LOG.info(f"Deleting all relational encounter data for encounter {encounter_id}")
+
+    # Take a snapshot of relational location data that is safe to delete before we delete our encounter.
+    location_ids_unique_to_this_encounter = db.fetch_all(
+        f"""
+          SELECT
+            location_id
+          FROM
+            warehouse.encounter_location t1
+          WHERE
+            encounter_id = {encounter_id}
+            AND location_id NOT IN
+              (
+                SELECT
+                  location_id
+                FROM
+                  warehouse.encounter_location
+                WHERE
+                  location_id = t1.location_id
+                  AND encounter_id <> {encounter_id}
+              )
+        """
+    )
+
+    deleted_encounter_loc_rows = delete_encounter_location(db, encounter_id)
+    LOG.debug(
+        f"Deleted {deleted_encounter_loc_rows} rows from `encounter_location` associated "
+        + f"with `encounter_id` = {encounter_id}"
+    )
+
+    sample_ids_associated_with_this_encounter = db.fetch_all(
+        f"SELECT encounter_id, sample_id FROM warehouse.sample WHERE encounter_id = {encounter_id}"
+    )
+    for sample in sample_ids_associated_with_this_encounter:
+        deleted_pa_rows = delete_presence_absence_by_sample(db, sample.sample_id)
+        deleted_sample_rows = delete_sample(db, sample.sample_id)
+        LOG.debug(
+            f"Deleted {deleted_pa_rows} rows from `presence_absence` and {deleted_sample_rows} from `sample` associated "
+            + f"with `encounter_id` = {sample.encounter_id} and `sample_id` = {sample.sample_id}"
+        )
+
+    # All relational encounter data with FK constraints has now been deleted, so it's safe to delete the encounter.
+    deleted_encounter_rows = delete_encounter(db, encounter_id)
+    LOG.debug(
+        f"Deleted {deleted_encounter_rows} rows from `encounter` associated "
+        + f"with `encounter_id` = {encounter_id}"
+    )
+
+    # Try to delete the individual associated with this encounter. We don't mind if this deletion fails,
+    # it just means that the individual is associated with more than one encounter.
+    try:
+        with db.savepoint(f"individual {individual_id}"):
+            deleted_individual_rows = delete_individual(db, individual_id)
+            LOG.debug(
+                f"Deleted {deleted_individual_rows} rows from `individual` associated with "
+                + f"`encounter_id` = {encounter_id} and `individual_id` = {individual_id}"
+            )
+    except ForeignKeyViolation:
+        LOG.debug(
+            f"Couldn't delete `individual_id` = {individual_id}. This individual_id is still referenced by other encounters in ID3C."
+        )
```

Comment:
Similar to locations, you could write a query for "individuals_unique_to_this_encounter" and only attempt to delete those.